### PR TITLE
Make WorkDetails component more flat

### PR DESCRIPTION
**Pull request by jamesgorrie**
## Who is this for?
Devs trying to work with `WorkDetails`

## What is it doing for them?
Makes the component more "flat" in the sense of it just uses this `{condition && <Component />}` pattern to render.

### CODE REVIEW

File: catalogue/webapp/components/WorkDetailsLinks/WorkDetailsLinks.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// @flow
+import NextLink from 'next/link';
+import Space from '@weco/common/views/components/styled/Space';
+import { classNames } from '@weco/common/utils/classnames';
+import WorkDetailsProperty from '../WorkDetailsProperty/WorkDetailsProperty';
+
+// TODO: Fix links type
```
Comment (davidpmccormick) [inline catalogue/webapp/components/WorkDetailsLinks/WorkDetailsLinks.js]: do we not have a `Link` type already, with `url` and `text` keys?

File: catalogue/webapp/pages/download.js [hunk 2]
```diff
@@ -106,12 +106,10 @@ const DownloadPage = ({ workId, sierraId, manifest, work }: Props) => {
             <SpacingComponent>
               <div id="licenseInformation">
                 <MetaUnit
```
Comment (gestchild) [inline catalogue/webapp/pages/download.js]: Looks like this should be the new <WorkDetailsText> component

File: catalogue/webapp/components/WorkDetailsText/WorkDetailsText.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// @flow
+
+type Props = {| title: ?string, text: string[] |};
+
+const WorkDetailsTags = ({ title, text }: Props) => {
```
Comment (davidpmccormick) [inline catalogue/webapp/components/WorkDetailsText/WorkDetailsText.js]: I think this should be `WorkDetailsText`

File: catalogue/webapp/components/WorkDetailsLinks/WorkDetailsLinks.js (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/components/WorkDetailsLinks/WorkDetailsLinks.js]: Copy / paste

**Pull request merged**